src/StringUtils.cpp

```diff
@@ -14,6 +14,223 @@
+{
+	return detail::rangeCheckHelper<unsigned short>(value);
+}
+
+template<>
+int NAS2D::stringTo(const std::string& value)
+{
+	return std::stoi(value);
+}
+
+template<>
+unsigned int NAS2D::stringTo(const std::string& value)
+{
+	const auto valueAsInteger = std::stoul(value);
+	if constexpr (std::numeric_limits<unsigned int>::max() < std::numeric_limits<unsigned long>::max())
+	{
+		if (std::numeric_limits<unsigned int>::max() < valueAsInteger)
+		{
+			throw std::out_of_range("from_string argument out of range");
+		}
+	}
+	return static_cast<unsigned int>(valueAsInteger);
+}
+
+template<>
+long NAS2D::stringTo(const std::string& value)
+{
+	return std::stol(value);
+}
+
+template<>
+unsigned long NAS2D::stringTo(const std::string& value)
+{
+	return std::stoul(value);
+}
+
+template<>
+long long NAS2D::stringTo(const std::string& value)
+{
+	return std::stoll(value);
+}
+
+template<>
+unsigned long long NAS2D::stringTo(const std::string& value)
+{
+	return std::stoull(value);
+}
+
+template<>
+float NAS2D::stringTo(const std::string& value)
+{
+	return std::stof(value);
+}
+
+template<>
+double NAS2D::stringTo(const std::string& value)
+{
+	return std::stod(value);
+}
+
+template<>
+long double NAS2D::stringTo(const std::string& value)
+{
+	return std::stold(value);
+}
+
+
+template<>
+std::string NAS2D::stringFrom(std::string value)
+{
+	return value;
+}
+
+template<>
+std::string NAS2D::stringFrom(bool value)
+{
+	return value ? "true" : "false";
+}
+
+template<>
+std::string NAS2D::stringFrom(char value)
+{
+	return std::string{value};
+}
+
+template<>
+std::string NAS2D::stringFrom(signed char value)
+{
+	return std::to_string(static_cast<int>(value));
+}
+
+template<>
+std::string NAS2D::stringFrom(unsigned char value)
+{
+	return std::to_string(static_cast<unsigned int>(value));
+}
+
+template<>
+std::string NAS2D::stringFrom(short value)
+{
+	return std::to_string(static_cast<int>(value));
+}
+
+template<>
+std::string NAS2D::stringFrom(unsigned short value)
+{
+	return std::to_string(static_cast<unsigned int>(value));
+}
+
+template<>
+std::string NAS2D::stringFrom(int value)
+{
+	return std::to_string(value);
+}
+
+template<>
+std::string NAS2D::stringFrom(unsigned int value)
+{
+	return std::to_string(value);
+}
+
+template<>
+std::string NAS2D::stringFrom(long value)
+{
+	return std::to_string(value);
+}
+
+template<>
+std::string NAS2D::stringFrom(unsigned long value)
+{
+	return std::to_string(value);
+}
+
+template<>
+std::string NAS2D::stringFrom(long long value)
+{
+	return std::to_string(value);
+}
+
+template<>
+std::string NAS2D::stringFrom(unsigned long long value)
+{
+	return std::to_string(value);
+}
+
+template<>
+std::string NAS2D::stringFrom(float value)
+{
+	return std::to_string(value);
+}
+
+template<>
+std::string NAS2D::stringFrom(double value)
+{
+	return std::to_string(value);
+}
+
+template<>
+std::string NAS2D::stringFrom(long double value)
+{
+	return std::to_string(value);
+}
```

Comment:
These all have the exact same implementation, and can be collapsed to a single definition.

Reply:
Could you elaborate?

Reply:
```cpp
template<typename T>
std::string NAS2D::stringFrom(T value)
{
	return std::to_string(value);
}
```

Reply:
If you want to get fancy, you can use SFINAE to limit the definition to types for which `std::to_string` is callable with that type. The SFINAE expression might look something like:
```cpp
decltype(std::to_string(declval<T>()))
```

I'm not quite certain on the exact syntax needed, though it might look something like:
```cpp
template<typename T>
decltype(std::to_string(declval<T>()))
NAS2D::stringFrom(T value)
{
	return std::to_string(value);
}
```

That should result in a return type of `std::string`, provided `std::to_string` is callable with the template type.